app/scripts/services/DataRepository.js

```diff
@@ -190,12 +204,40 @@ console.log('Success...');
         }
       },
       function (error) {
-console.log('Error...', error);
         window.alert('Error: ' + error.toString());
       }
     );
   };
 
+  this.populateRecordFromBrandModel = function populateRecordFromBrandModel(brandRecord, brand) {
+    brandRecord.name = brand.getName();
+    brandRecord.description = brand.getDescription();
+    brandRecord.categories = brand.getCategoryIdAsCsl();
+    brandRecord.image = brand.getImage();
+    brandRecord.ranking = brand.getRanking();
+    // brandRecord.brandUrl = brand.getBrandURL(); // @TODO: There does not seem to be a URL property in the data.
+  };
+
+  this.populateRecordFromCategoryModel = function populateRecordFromCategoryModel(categoryRecord, category) {
+    categoryRecord.description = category.getDescription();
+    categoryRecord.image = category.getImage();
+    categoryRecord.name = category.getName();
+    categoryRecord.parentCategoryId = category.getParentCategoryId();
+  };
+
+  this.populateRecordFromProductModel = function populateRecordFromProductModel(productRecord, product) {
+    productRecord.brandId = product.getBrandId();
+    productRecord.categoryId = product.getCategoryId();
+    productRecord.description = product.getDescription();
+    productRecord.id = product.getId();
+    productRecord.image = product.getImage();
+    productRecord.name = product.getName();
+    productRecord.parentCategoryId = product.getParentCategoryId();
+    productRecord.purchaseURlClicks = product.getPurchaseUrlClicks();
+    productRecord.purchaseUrl = product.getPurchaseUrl();
+    // productRecord.$$hashKey = product.$$hashKey(); @TODO: Do we save this? When does it change?
```

Comment:
My understanding was that this was some internal key to firebase that we don't need to worry about (still very worth looking into, because I'm not sure I'm remembering that correctly or if it's just wishful thinking 😄).

Reply:
Haha! Wishful thinking it a dangerous thing!!
I'll leave it for now, but make a note to look into it. Is this a feature of firebase (ie does it generate this hash itself?)